x-pack/plugins/ml/public/embeddables/anomaly_charts/types.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MlEntityField } from '@kbn/ml-anomaly-utils';
+import type { HasType, PublishingSubject } from '@kbn/presentation-publishing';
+import type { JobId } from '../../shared';
+import type { AnomalyExplorerChartsEmbeddableType } from '../constants';
+import type { MlEmbeddableBaseApi } from '../types';
+
+export interface AnomalyChartsFieldSelectionApi {
+  jobIds: PublishingSubject<JobId[]>;
+  entityFields: PublishingSubject<MlEntityField[] | undefined>;
+}
+
+export type AnomalyChartsEmbeddableApi = HasType<AnomalyExplorerChartsEmbeddableType> &
```

Comment:
This type is really clean and straightforward. Nice work!

This is not necessary, but interfaces have slightly better editor time performance than types, so we're starting to recommend using the `interface` syntax whenever possible. This could be rewritten like:

``` typescript
export interface AnomalyChartsEmbeddableApi extends HasType<AnomalyExplorerChartsEmbeddableType>, MlEmbeddableBaseApi, AnomalyChartsFieldSelectionApi {}
```

Reply:
Updated in https://github.com/elastic/kibana/pull/179073/commits/517e8a8ea331da1c08ece6919ff2f89bda132032